Mantle dynamics l0

alarshi opened the pull request:
This PR follows up from PR #10 to add the mantle dynamics notebook.

JarettBakerDunn committed:
add tectonics-modeling directory and aspect day 1 tutorial

Commit:
Start breaking down the notebook into constituents.

alarshi committed:
First draft of the mantle dynamics notebook.

alarshi committed:
First draft.

alarshi committed:
Add the parameter file.

alarshi committed:
Addressed suggestions from drive and text updates.

ljhwang commented on bin/mantle-dynamics-L0/1-1_understanding_the_parameter_file.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"1a7e3f50","metadata":{"id":"1a7e3f50"},"source":["## The .prm file\n","\n","For this module, you will need the parameter file 'mantle-flow.prm' which is available in the current directory. All the input parameters in this file can be loaded as a dictionary using the python script, `aspect_prmf.py` available [here](https://github.com/mgouiza/aspect-prmf).\n" ...
```
<p>Global Parameters is mispelled</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1989928565'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang replied:
<p>should we be using punctution here? NOTE vs NOTE:</p><p>Add a period at the end of this sentence too.</p><p> </p><p>Global Paramters -&gt; Global Parameters</p><p>... to describe the simulation parameters that determine the overall behavior of the simulation such as the time duration and the output directory to store the solution files.</p><p> </p><p>change comma at the end of the sentence to a period: ...left and right temperature options can be similarly specified, In 3D models,&nbsp;</p><p>they -&gt; the When boundaries are unspecified, they default</p><p> </p><p>Velocity at the Boundary - Are we removing commas from bullet lists?</p><p> </p><p>Is "adaptive" capitalized -&gt;"Adaptive refinement"</p><p> </p><p>spelling: material models from whcih the - &gt; material models from which the</p><p>remove extra space: which depends on temperature as : -&gt; which depends on temperature as:</p><p>fix math . should be: ... where, $\rho_0$, $\alpha$, $T$, ...</p><p> </p><p>Should there be a link to the online manual if years want to investigate the parameters further?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996268255'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang replied:
<p>The next section is plate tecttonics and not Convection in a box model.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996268260'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

ljhwang commented on bin/mantle-dynamics-L0/0_overview_mantle_dynamics_L0.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"bfec1c83-9384-4796-902c-d750d13f348b","metadata":{"id":"bfec1c83-9384-4796-902c-d750d13f348b"},"source":["<br>\n","\n","\n","<table style=\"width: 70%; border-collapse: collapse; border: none; margin-left:60px\">\n","  <tr>\n","    <td style=\"width: 20%\" align=\"center\"> <img src=\"../assets/education-gem-notebooks_icon.png\" alt=\"icon\"  width=\"80%\"  ...
```
<p>We probably should enforce some consistency in the bullet lists. Should they all start with a capital letter? AI says no commas "," as well.</p><p> </p><p>Add a Dependencies header for completeness and just in case we might need to add later.</p><p> </p><p>scipy - &gt; SciPy</p><p>matplotlib -&gt; Matplotlib</p><p>anlysis -&gt; analysis</p><p> </p><p>add "the" - Fundamentals of heat transport mechanisms inside the Earth</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

ljhwang replied:
<p>paramter - &gt; parameter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

ljhwang commented on bin/mantle-dynamics-L0/0_overview_mantle_dynamics_L0.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"bfec1c83-9384-4796-902c-d750d13f348b","metadata":{"id":"bfec1c83-9384-4796-902c-d750d13f348b"},"source":["<br>\n","\n","\n","<table style=\"width: 70%; border-collapse: collapse; border: none; margin-left:60px\">\n","  <tr>\n","    <td style=\"width: 20%\" align=\"center\"> <img src=\"../assets/education-gem-notebooks_icon.png\" alt=\"icon\"  width=\"80%\"  ...
```
<p>add "in:" -&gt; in 2020 led </p><p> </p><p>add "original" -&gt; The original slides</p><p> </p><p>github -&gt; GItHub</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

alarshi replied:
I think the original draft of the notebook by Jarett was indeed based on the tectonics modeling workshop, but now we have modified it quite a lot to demonstrate convection in a system. I incorporated your changes and modified the original sentence 'Course materials are based on the tectonics' -> 'Course materials are modified from the tectonics'.

ljhwang commented on bin/mantle-dynamics-L0/2_plate_tectonics_and_basic_equations.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"bf22d7a6","metadata":{"id":"bf22d7a6"},"source":["## Plate tectonics\n","\n","Plate tectonics can explain :\n"," * how plates move\n"," * provide a basic explanation for mountains/basins, volcanoes and earthquakes\n","\n","The illustration below shows mantle convection cell and the associated plate tectonics :\n","\n","<figure>\n","<img src=\"images/mantle_ ...
```
<p>Extra spaces before colon throughout.</p><p> </p><p>The resitance between ... - &gt; The resistance between ...</p><p> </p><p>Add a period at the end of a complete sentence: This suction pulls the overriding plate towards the trench</p><p> </p><p>Image has a typo. Either an paren is missing or there is an extra one  in "Slab pull (due to gravity"</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996285504'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang replied:
<p>remove spaces before colons</p><p> </p><p>Assuming mantle to&nbsp;...  -&gt; Assuming the mantle to&nbsp;...</p><p> </p><p>&nbsp;is also referred as&nbsp;<em>Navier-Stokes</em>&nbsp; -&gt; &nbsp;is also referred to as the&nbsp;<em>Navier-Stokes</em>&nbsp;</p><p> </p><p>how it evolved -&gt; how it evolves</p><p> </p><p>&nbsp;its density increases. - &gt; &nbsp;its density increases:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996285506'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

ljhwang commented on bin/mantle-dynamics-L0/3_convection_in_box_model.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"4f8dae51","metadata":{"id":"4f8dae51"},"source":["## Running the model\n","\n","To run ASPECT using the .prm file that decribes a 2D box heated at the bottom (more details in the [previous notebook](./2_understanding_the_prm_file.ipynb)), execute the following line :"]},{"cell_type":"code","execution_count":null,"id":"b0e2cf93","metadata":{"id":"b0e2cf93"}, ...
```
<p>remove space before colon</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996295958'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang replied:
<p>aspect -&gt; ASPECT</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996295960'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

ljhwang replied:
<p>The last sentence seems awkward.  Suggested modification</p><p> </p><p>... we need to look at the Rayleigh and Nusselt numbers.</p><p>... we need to look at the Rayleigh Nusselt number relationship.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996295962'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

ljhwang commented on bin/mantle-dynamics-L0/4_Rayleigh_Nusselt_Relationship.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"4606b260","metadata":{"id":"4606b260"},"source":["## Rayleigh--Nusselt Relationship\n","\n","### Rayleigh Number\n","\n","The expected degree of convection is described by a dimensionless number, Rayleigh Number (Ra):\n","\n","\\begin{align}\n","    Ra = \\frac{\\rho_0 g {\\alpha}{\\Delta}T D^3}{{\\eta}k}\n","\\end{align}\n","\n","Here, $\\Delta \\rho$ is t ...
```
<p>Rayleigh Number -&gt; Rayleigh number</p><p> </p><p>There is no Here,&nbsp;Δρ in the above equation</p><p> </p><p>and temprature difference -&gt; and temperature difference</p><p> </p><p>I am not sure I like using bulleted lists here. Alternative:</p><ul><li>A cubic millimeter of honey will not convect for a temperature difference of 100K BUT a cubic kilometer of honey may.</li><li>A cubic kilometer of honey will not convect for a temperature difference of 0.1K BUT it may with a temperature difference of 100K</li></ul><p>And then delete the Also: ???</p><p> </p><p>Use more words for In summary. Suggestion:</p><p>In summary, as the Rayleigh number increase, convection becomes more vigorous, and more heat is transported convectively.</p><p> </p><p>"Nusselt Number" - &gt; The "Nusselt number"</p><p>missing space after period: boundary.The Rayleigh&nbsp;-&gt; boundary. The Rayleigh&nbsp;</p><p>Nusselt number, which is:&nbsp;N -&gt; Nusselt number:&nbsp;N</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996309998'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang replied:
<p>vs -&gt; vs.</p><p> </p><p>sometimes we refer to it as .prm and other times prm (without the dot). We should stick to one or the other. My preference is .prm.</p><p> </p><p>For example, halving the default value :  -&gt; For example, halve the default value:</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996310000'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

ljhwang commented on bin/mantle-dynamics-L0/5_different_tectonic_regimes.ipynb:
```diff
@@ -0,0 +1 @@
+{"cells":[{"cell_type":"markdown","id":"46148f74","metadata":{"id":"46148f74"},"source":["## Onset of Convection\n","\n","As described in the previous notebook, Rayleigh number (Ra), describes the vigor of convection as :\n","\n","\\begin{align}\n","    Ra = \\frac{\\rho_0 g {\\alpha}{\\Delta}T D^3}{{\\eta}k}\n","\\end{align}\n","\n","It is important to note that:\n","* below a critical value for ...
```
<p>...&nbsp;notebook, Rayleigh number (Ra), describes the vigor of convection as : -&gt; ... &nbsp;notebook, the Rayleigh number (Ra), describes the vigor of convection as:</p><p>complete sentence, add a period: </p><p>     In this case, heat is transferred purely by conduction.</p><p>     above this ciritcal value, heat is transported by convection</p><p>... above this ciritcal value ... -&gt; ... above this critical value ...</p><p> </p><p>Again, I am wondering if this is better as a sentence vs. bullet points.</p><p> </p><p>missing period at end of sentence:  ... for the line of onset of convection</p><p> </p><p>Not sure what the real point of this sentences is:</p><p>Convection is linked closely to the lithospheric plates, however, it is possible that even with a high enough Ra, some planets do not have mobile plates.</p><p>Rewrite as:</p><p>Convection is linked closely to the physical properties of the lithospheric plates. It is possible that even with a high Ra, the plates of some planets will not be mobile.</p><p> </p><p>... the planet's plates will not break and form boundaries with surface motion.</p><p> </p><p>On the other hand, some planetary bodies -&gt; Some planetary bodies </p><p>remove space before colon: ...tectonic regimes :</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geodynamics-hubzero/gem-notebooks/pull/12/discussion/#comment-1996321641'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

ljhwang commented:
The prm file section looks like it was uploaded twice under different file names. The one named 2_understanding_the_prm_file.ipynb should be deleted?

alarshi committed:
Address suggestions from the PR.

alarshi commented:
Thanks, Lorraine, for the careful review! :)  I addressed your suggestions in the latest push and removed the redundant notebook (`2_understanding_*.ipynb`).
